src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java

```diff
@@ -346,21 +347,58 @@ public void setWriteTimeout(Duration writeTimeout) {
 			return CompletableFuture.completedFuture(null);
 		}
 
+		/**
+		 * Writes the specified byte to this response stream.
+		 */
 		@Override
-		public void write(int b) {
-			this.writeAsync(new byte[] { (byte) b }, 0, 1).join();
+		public void write(int b) throws IOException {
+			try {
+				this.writeAsync(new byte[] { (byte) b }, 0, 1).join();
+			} catch (CompletionException e) {
+				throw new IOException(e.getCause());
+			}
 		}
 
-		public void write(byte[] bytes) {
-			this.writeAsync(bytes, 0, bytes.length).join();
+		/**
+		 * Writes b.length bytes from the specified byte array to this response stream.
+		 */
+		@Override
+		public void write(byte[] b) throws IOException {
+			try {
+				this.writeAsync(b, 0, b.length).join();
+			} catch (CompletionException e) {
+				throw new IOException(e.getCause());
+			}
```

Comment:
Can this simply be a one line call to the other **sync** overload?

this.write(b, 0, b.length);

Reply:
Good idea

---
In reply to: [293102083](https://github.com/Azure/azure-relay-java/pull/46#discussion_r293102083) [](ancestors = 293102083)